tools/Custom/ListCmdlet.cs

```diff
@@ -84,11 +101,8 @@ public bool ShouldIteratePages(global::System.Collections.Generic.Dictionary<str
         {
             iteratedPages++;
             totalFetchedItems += itemsCount;
-            if ((boundParameters.ContainsKey("All") && !boundParameters.ContainsKey("PageSize")) ||
-                (boundParameters.ContainsKey("PageSize") && totalFetchedItems < originalPageSize))
-                return true;
-            else
-                return false;
+
+            return limit == default || totalFetchedItems < limit;
```

Comment:
may be helpful to add some quick comments.  For example 'we iterate if there's no top/limit parameter or if there are more items to fetch'